### ISSUE TITLE
P2P Incentivization Model - censorship resistance 

### ISSUE THREAD

**Issue by rigelrozanski** [cosmos/cosmos-economics#11]
Jae, Sunny, + Adrian, Rigel, Gautier, Sunny

The more signatures a proposers includes in a block, the more fee rewards that proposer receives. 
They always at least 2/3 but could be up to 1/1 - out of the dispropostionate amount of surprus fee rewards that go to the proposer (small percent ~105%)

Reasons for decision: 
 - Validators now have an incentive to be strongly connected to every other validator. Since it will mean that they receive more signature for the block they are proposing. 
 - Proposers are disincentivize from censoring as they will get less fee rewards. 

Open Question - what is the distribution function between for including extra signatures.

Concerns: 
 - This incentizes proposers to wait for more signatures, increasing block times. This could maybe be solved by looking at a sliding window of signatures

**Comment by gamarin2** [comment]
So could we combine that with a punishment for unavailabaility ?

E.g. if a validator's signature is not in any block over a 1 day period he gets slashed for a little amount (1%), and if his signature is not in any block over a 1 week period he gets unbonded

**Comment by adrianbrink** [comment]
@gamarin2 Yes I think that is a good idea. I've created an issue with the ideas. Let's move the discussion to #12.

**Comment by ebuchman** [comment]
Another concern is it could cause proposers to fight - ie. folks will not vote for a proposal so that they can try to be the proposer.

> So could we combine that with a punishment for unavailabaility ?

I thought rewarding the proposer is supposed to be instead of punishing for "unavailability"? Or was it discussed that there should be both?

**Comment by gamarin2** [comment]
> I thought rewarding the proposer is supposed to be instead of punishing for "unavailability"? Or was it discussed that there should be both?

Rewarding the proposer is to incentivize better networking and disincentivize censorship. We still need a way to punish for unavailability and eventually unbond inactive validators.

**Comment by sunnya97** [comment]
If we include the data from pre-votes as well, not just the pre-commits, we can get more data about the live validators than from just the pre-commits alone.  For example, imagine 4 validators:  A, B, C, and D.  A is the next proposer.  He only needs 3 pre-commits to propose the next block.  Let's say he has his own, B's, and C's.  This would incentivize him to wait for D's pre-commit as well.  But what if B's pre-commit has data acknowledging D's pre-vote.  The protocol can recognize that, and that way A doesn't have to wait for a pre-commit from D.  I believe this is more intuitive for a peer-to-peer mesh network layout.

**Comment by rigelrozanski** [comment]
>I thought rewarding the proposer is supposed to be instead of punishing for "unavailability"? Or was it discussed that there should be both?

yeah @gamarin2 that makes sense. More on this... YES we do need both, but the two mechanisms work hand in hand. If a validator becomes unavailable for a large period of time the network should not wait for them to catch up (example, validators being destroyed in war), but to enforce the fact that this incentivizes validators to attempt to "kick" each other offline we introduce this slight incentive mechanism to promote publishing evidence of all validators that they see as still online.
